### PR TITLE
[Sample] [Graphics] UltraHDR x OpenGL shaders with SurfaceView brightness control

### PR DESCRIPTION
**Pull request by madebymozart**
# Description
This PR uses UltraHDR image, and processes them through a OpenGL pipeline using fragment & vertex shaders. This then displays the output frame on the SurfaceView and shows the developer how to control the brightness of the UltraHDR Image.

The included fragment shader is a tone-mapping shader. Big thanks for @johnreck for the graphics code here!

# How Has This Been Tested?
**Test Configuration #1 - Pixel 8 Pro API 34 Beta (Android 14)**
**Test Configuration #1 - Pixel 7 Pro API 34 Beta (Android 14 Beta)**

# Demo
Screenshot_20231206_124445

# Recording
https://github.com/android/platform-samples/assets/5649571/e199f896-92fc-4f6c-9fea-5d6a657058e4

### CODE REVIEW

File: samples/graphics/ultrahdr/src/main/res/layout/ultrahdr_with_graphics.xml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (yaraki) [reply samples/graphics/ultrahdr/src/main/res/layout/ultrahdr_with_graphics.xml:2]: It seems shader files got copyright notices, but not this XML file.

[the remark this answers is not in the document]
Comment (madebymozart) [reply samples/graphics/ultrahdr/src/main/res/layout/ultrahdr_with_graphics.xml:2]: Ahh must have missed that one, adding now

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -124,6 +124,7 @@ androidx-work-runtime-ktx = "androidx.work:work-runtime-ktx:2.8.1"
 androidx-core-remoteviews = "androidx.core:core-remoteviews:1.0.0"
 androidx-glance-appwidget = "androidx.glance:glance-appwidget:1.0.0"
 androidx-glance-material3 = "androidx.glance:glance-material3:1.0.0"
+androidx-graphics-core = { group = "androidx.graphics", name = "graphics-core", version = "1.0.0-alpha05" }
```
Comment (nadewad) [inline gradle/libs.versions.toml:127]: Note as of Dec 13th the 1.0.0-beta01 release of the graphics-core library is now available.

File: samples/graphics/ultrahdr/src/main/java/com/example/platform/graphics/ultrahdr/opengl/UltraHDRWithOpenGL.kt [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    override fun onDrawComplete(
+        targetSurfaceControl: SurfaceControlCompat,
+        transaction: SurfaceControlCompat.Transaction,
+        frameBuffer: FrameBuffer,
+        syncFence: SyncFenceCompat?,
+    ) {
+        transaction.setDataSpace(targetSurfaceControl, if (isWideGamut) P3_XRB else SRGB_XRB)
+        transaction.setExtendedRangeBrightness(targetSurfaceControl, hdrSdrRatio, desiredRatio)
+        transaction.commit()
+        super.onDrawComplete(targetSurfaceControl, transaction, frameBuffer, syncFence)
```
Comment (nadewad) [inline samples/graphics/ultrahdr/src/main/java/com/example/platform/graphics/ultrahdr/opengl/UltraHDRWithOpenGL.kt:242]: Just FYI, delegation to the base class method is not necessary.

Also the implementation of GLFrameBufferRenderer already calls transaction.commit on your behalf so explicitly calling commit here is not necessary.

Comment (madebymozart) [reply samples/graphics/ultrahdr/src/main/java/com/example/platform/graphics/ultrahdr/opengl/UltraHDRWithOpenGL.kt:242]: 👍🏾 Perfect - Thanks for letting me know

File: gradle/libs.versions.toml (hunk 2 above)
Comment (madebymozart) [reply gradle/libs.versions.toml:127]: Updated it in latest patch

**Review by yaraki — APPROVED** [review]

**Pull request merged**